module/zfs/dsl_scan.c

```diff
@@ -682,6 +683,10 @@ dsl_scan_setup_check(void *arg, dmu_tx_t *tx)
 	if (dsl_scan_is_running(scn))
 		return (SET_ERROR(EBUSY));
 
+	if (dmu_tx_is_syncing(tx) && scn->scn_restart_txg != 0 &&
```

Comment:
Can you add a quick comment explaining what this `if()` block is checking for?

Reply:
this is a rescheduling of SCRUB if an 'attach/add' in progress.